src/persist/src/indexed/columnar/arrow.rs

```diff
@@ -132,14 +153,14 @@ fn realloc_data(data: ArrayData, metrics: &ColumnarMetrics) -> ArrayData {
 /// Re-allocate the backing storage for a specific array using lgalloc, if it's configured.
 pub fn realloc_array<A: Array + From<ArrayData>>(array: &A, metrics: &ColumnarMetrics) -> A {
     let data = array.to_data();
-    let data = realloc_data(data, metrics);
+    let data = realloc_data(data, true, metrics);
```

Comment:
Is it safe to pass `true` initially because the upstream Parquet issue only occurs for nested arrays, and this method is called on the top-level array?

Do you also mind adding a comment with an explanation for passing `true` since I know I'll forget in a month :)

Reply:
Yeah, that's right! (And in practice arrow only lets you restrict nullability for fields, not top-level arrays.) I'll add the comments.